guides/api-design.md

```diff
@@ -0,0 +1,141 @@
+# API Design Guidelines
+
+* Authors: Mukundan Sundararajan (@mukundansundar), John Ewart (@johnewart)
+* Updated: 10/25/2022
+
+## Proposal requirements
+
+
+For any API (new or updates), the following must be included in the proposal:
+
+  * Relevant high level design
+  * Proposed contract for the API
+    * HTTP and gRPC APIs should be consistent in behavior and user experience.
+  * Identifying what additions to existing components / creation of new components are required for this API (if any)
+  * Scope for current and following releases (i.e what can be expected from this iteration and what is being pushed down the road)
+  * Known limitations, where applicable
+    * Performance issues
+    * Compatibility issues
+  * Code examples (pseudocode is acceptable)
+
+
+## API Lifecycle expectations
+
+APIs are expected to go through three stages in their lifetime: Alpha, Beta and Stable. For each of these phases it should be clear to a user what they can expect. In the case of an API, those expectations are:
+
+* **Alpha**
+   * API is not production ready yet and might contain bugs
+   * Recommended for non-business-critical use only because of potential for incompatible changes in subsequent releases
+   * May not be backwards-compatible with an API it intends to replace
+   * May not be highly performant or support all SDKs
+
+* **Beta**
+   * API is not production ready yet
```

Comment:
Mention: API is intended to move to stable and not be deleted.

Reply:
Added